test/OpenTelemetry.Instrumentation.W3cTraceContext.Tests/W3CTraceContextTests.cs

```diff
@@ -57,9 +64,89 @@ public void W3CTraceContextTestSuite(string value)
                 string lastLine = ParseLastLine(result);
                 this.output.WriteLine("result:" + result);
                 Assert.StartsWith("FAILED", lastLine);
+#if NETCOREAPP2_1
+                AssertOutputOfNetCoreApp21(result);
+#else
+                AssertOutputOfNetCoreApp31(result);
+#endif
+            }
+        }
+
+        private static void AssertOutputOfNetCoreApp21(string output)
+        {
+            string[] unsuccessfulTestcaseNames = ExtractUnsuccessfulTestcaseNames(output);
+
+            // TODO: fix and remove from this failure list except for known failures with external dependencies. #1219
+            // See failures with external dependencies in https://github.com/open-telemetry/opentelemetry-dotnet/issues/404
+            string[] existingFailures = new string[]
+            {
+                // Errors:
+                "test_tracestate_duplicated_keys",
+                "test_tracestate_key_illegal_characters",
+                "test_tracestate_key_illegal_vendor_format",
+                "test_tracestate_key_length_limit",
+                "test_tracestate_member_count_limit",
+                "test_tracestate_multiple_headers_different_keys",
+                "test_tracestate_value_illegal_characters",
+
+                // Failures:
+                "test_traceparent_trace_flags_illegal_characters",  // External dependency. See issue #404
+                "test_traceparent_version_0x00",
+                "test_traceparent_version_0xff",
+                "test_tracestate_empty_header",
+            };
+
+            // FAILED (failures=4, errors=7)
+            Assert.Equal(existingFailures.Length, unsuccessfulTestcaseNames.Length);
+            for (int i = 0; i < existingFailures.Length; ++i)
+            {
+                Assert.Equal(existingFailures[i], unsuccessfulTestcaseNames[i]);
+            }
+        }
+
+        private static void AssertOutputOfNetCoreApp31(string output)
```

Comment:
Guard this definition by `#if NETCOREAPP3_1`?

Reply:
I think your intention is to remove the unused code warning? I've moved the `#if` statement inside of this method to avoid that.